cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/CentralOAuthServiceImpl.java

```diff
@@ -288,29 +293,23 @@ public Boolean revokeToken(final Token token) {
 
     @Override
     public Boolean revokeClientTokens(final String clientId, final String clientSecret) {
-        final OAuthRegisteredService service = getRegisteredService(clientId);
-        if (service == null) {
-            LOGGER.error("OAuth Registered Service could not be found for clientId : {}", clientId);
-            return Boolean.FALSE;
-        }
-        if (!service.getClientSecret().equals(clientSecret)) {
-            LOGGER.error("Invalid client secret");
-            return Boolean.FALSE;
-        }
 
         final Collection<RefreshToken> refreshTokens = tokenRegistry.getClientTokens(clientId, RefreshToken.class);
-        for (final RefreshToken token : refreshTokens) {
-            LOGGER.debug("Revoking refresh token : {}", token.getId());
-            ticketRegistry.deleteTicket(token.getTicket().getId());
```

Comment:
### This Issue
If there is more than one token, the second execution fails here. Here is a [link](https://github.com/apereo/cas/blob/4.1.x/cas-server-support-jdbc/src/main/java/org/jasig/cas/ticket/registry/JpaTicketRegistry.java#L96) to the failure code that throws the following error:
```
object references an unsaved transient instance - save the transient instance before flushing
```
Please note this happens for the access token below as well.

The first attempt is to  update our `AccessTokenImpl` to set `CascadeType.ALL` for its property `TicketGrantingTicket, which prevents the error but does not update the database.

Reply:
Update, further investigation reveals that this is caused by the `@transcient` annotation [link](https://github.com/CenterForOpenScience/cas-overlay/blob/develop/cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/token/AccessTokenImpl.java#L96):

```java
    @Transient
    public Ticket getTicket() {
        if (getType() == TokenType.OFFLINE) {
            return this.serviceTicket;
        }

        return this.ticketGrantingTicket;
    }
```